agent/consul/state/catalog.go

```diff
@@ -1236,7 +1236,11 @@ func (s *Store) Services(ws memdb.WatchSet, entMeta *acl.EnterpriseMeta, peerNam
 	for service := services.Next(); service != nil; service = services.Next() {
 		result = append(result, service.(*structs.ServiceNode))
 	}
-	return idx, result, nil
+	parsedResult, err := parseServiceNodes(tx, ws, result, entMeta, peerName)
+	if err != nil {
+		return 0, nil, fmt.Errorf("failed querying and parsing services :%s", err)
+	}
+	return idx, parsedResult, nil
```

Comment:
@rboyer I saw you were the one that added `parseServiceNodes` to a bunch of other endpoints back in 2022. Is there any performance implications we need to think about for adding this to this endpoint as well? This change is needed to fill in nodemeta so it can be filtered on.

Reply:
Currently we only do the expensive join against the `nodes` table when the user provides the NodeMeta parameter on the original API query. When they do this we call a completely different function [link](https://github.com/hashicorp/consul/blob/main/agent/consul/catalog_endpoint.go#L596).

If we were concerned like that, we could instead alter the `Store.Services` method to take an extra boolean parameter to _conditionally_ join the NodeMeta as in this PR.

Then the calling code in `agent/consul/catalog_endpoint.go` could set the flag to `true` when `filter != ""` OR just when:
```
strings.Contains(strings.ToLower(filter), "nodemeta")
```

Reply:
@rboyer I have updated the PR can I merge this now? please approve.

Reply:
@rboyer I have updated the PR can I merge this now? please approve.